sql/database.sql

```diff
@@ -2662,9 +2662,9 @@ DROP TABLE IF EXISTS `lang_custom`;
 CREATE TABLE `lang_custom` (
   `lang_description` varchar(100) NOT NULL default '',
   `lang_code` char(2) NOT NULL default '',
-  `constant_name` mediumtext NOT NULL default '',
-  `definition` mediumtext NOT NULL default ''
-) ENGINE=MyISAM ;
+  `constant_name` mediumtext NOT NULL,
+  `definition` mediumtext
```

Comment:
MySQL 5 does not allow to set default value to fields of certain types.
http://dev.mysql.com/doc/refman/5.7/en/data-type-defaults.html

It generates a warning and omits DEFAULT clause, so NOT NULL table becomes a null.

Frankly speaking, it is a bug not related to InnoDB.

I think that all TEXT NOT NULL items should be changed , but this change requires code inspection.